docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Async Processing
+---
+import {Icons} from '/src/components/icons'
+
+:::info
+
+This page covers the Responsive Async Processor: its benefits, its limitations, how to use it, and how to tune it.
+
+The Async Processor is available with Responsive client **v0.24.0** onward.
+:::
+
+## Why use the Async Processor
+Async processing allows Kafka Streams applications to concurrently process records within a single task, thus changing the upper limit of parallelization from the level of a Kafka partition to the level of a record key. 
+
+It works by giving each StreamThread a dedicated thread pool which it uses to process the records passing through an async processor. Records of the same key are still processed in order and all the same consistency and correctness guarantees of Kafka Streams are maintained, including exactly-once semantics.
+
+While async processing can help improve performance of any application with a large keyspace, it was built to shine when processors include a long-running step such as remote state store requests or generic RPCs. The classic solution for achieving high throughput when you are bottlenecked by network latency is to increase the parallelism of the calls to the target system, which is exactly what the async processor does for Kafka Streams. 
+
+This makes the Responsive Async Processor the perfect tool to complement Responsive state stores, meaning you don’t need to sacrifice performance in order to gain the availability and elasticity benefits of remote storage with Kafka Streams!
```

Comment:
Have we settled on "Responsive Async Processor" as a title?

(not saying I don't like the term, just wondering if we've actively decided to commit to this. If not, we probably shouldn't capitalize it -- or should just figure out what we want to call it. FWIW I do think "Responsive Async Processor" is fine, as long as everyone else likes it)

Reply:
I think we should drop "Responsive" if it's on the Responsive site since in that context it's obvious that it's Responsive's processor. (c.f. https://brand.hashicorp.com/product_logos)

Reply:
Agreed. I'll drop the 'Responsive' prefix.